src/core/core.scale.js

```diff
@@ -715,32 +718,26 @@ module.exports = function(Chart) {
 
 					labelX = isLeft ? me.right - labelXOffset : me.left + labelXOffset;
 
-					var yLineValue = me.getPixelForTick(index); // xvalues for grid lines
+					var yLineValue = me.getPixelForTick(index); // xvalues for ticks
 					yLineValue += helpers.aliasPixel(lineWidth);
 					labelY = me.getPixelForTick(index, gridLines.offsetGridLines) + optionTicks.labelOffset;
 
 					tx1 = xTickStart;
 					tx2 = xTickEnd;
-					x1 = chartArea.left;
-					x2 = chartArea.right;
-					ty1 = ty2 = y1 = y2 = yLineValue;
+					ty1 = ty2 = yLineValue;
 				}
 
 				itemsToDraw.push({
 					tx1: tx1,
 					ty1: ty1,
 					tx2: tx2,
 					ty2: ty2,
-					x1: x1,
-					y1: y1,
-					x2: x2,
-					y2: y2,
 					labelX: labelX,
 					labelY: labelY,
-					glWidth: lineWidth,
-					glColor: lineColor,
-					glBorderDash: borderDash,
-					glBorderDashOffset: borderDashOffset,
+					tmWidth: lineWidth,
+					tmColor: lineColor,
+					tmBorderDash: borderDash,
+					tmBorderDashOffset: borderDashOffset,
 					rotation: -1 * labelRotationRadians,
 					label: label,
```

Comment:
I would force a conversion of label to string: `label: '' + label`, then we can test `if (label)` for empty value.